62.CAD/vertex_shader.hlsl

```diff
@@ -2,18 +2,213 @@
 
 #include "common.hlsl"
 
+float2 Tangent(float2 p0, float2 p1, float2 p2, float t)
+{
+    float2 tangent = 2.0 * (1.0 - t) * (p1 - p0) + 2.0 * t * (p2 - p1);
+    return tangent;
+}
+
+float TforTan(float2 p0, float2 p1, float2 p2, float2 DirNormalized)
```

Comment:
name of this function is not descriptive at all, if you can't come up with a better and longer name, add comments